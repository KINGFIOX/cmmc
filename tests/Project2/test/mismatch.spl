struct test2 {
    int x;
};
int foo(int x) {
    return 1;
}
int foo2(int x[2]) {
    return x[0];
}
struct test2 foo3(struct test2 x[2]) {
    return x[0];
}
int foo4(int x) {
    return 1.0;
}
int main() {
    int x = 0;
    float y1 = foo(1);
    float y;
    int z[10];

    foo();
    foo(y);
    y = -foo(x);
    foo('x');
    z[y] = z[x] + 
    z['x'];
    foo2(z);
    foo3(z);
    (x = x) = x;
    x(10);
    if(0.0) {
        if(0) {
        }
    }    

    return 0;
}
