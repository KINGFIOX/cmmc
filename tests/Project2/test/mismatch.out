Error type 8 at Line 14: incompatiable return type
Error type 5 at Line 18: unmatching type on both sides of assignment
Error type 9 at Line 22: invalid argument number for foo, expect 1, got 0
Error type 9 at Line 23: unmatching type for function call
Error type 5 at Line 24: unmatching type on both sides of assignment
Error type 9 at Line 25: unmatching type for function call
Error type 12 at Line 26: indexing by non-integer
Error type 12 at Line 27: indexing by non-integer
Error type 9 at Line 28: unmatching type for function call
Error type 9 at Line 29: unmatching type for function call
Error type 6 at Line 30: left side in assignment is rvalue
Error type 11 at Line 31: invoking non-function variable: x
Error type 4294967295 at Line 32: unmatching type for cmmc::ConversionUsage::Condition
